src/processor/snapshot/mod.rs

```diff
@@ -1,74 +1,49 @@
-use std::{collections::HashMap, fmt, fs, path::PathBuf, str::FromStr};
+use std::{fs, path::PathBuf, str::FromStr};
 
 mod bytecode;
+mod database;
 mod types;
 
 use async_trait::async_trait;
 use blake2::{Blake2s256, Digest};
 use ethers::types::{Address, H256, U256, U64};
 use eyre::Result;
-use indexmap::IndexSet;
 use state_reconstruct_fetcher::{
     constants::{ethereum, storage},
     types::CommitBlockInfoV1,
 };
 use tokio::sync::mpsc;
 
-use self::types::{SnapshotFactoryDependency, SnapshotStorageLog, StorageKey, StorageValue};
+use self::{
+    database::{DatabaseError, SnapshotDB},
+    types::{SnapshotFactoryDependency, SnapshotStorageLog},
+};
 use super::Processor;
-use crate::processor::snapshot::types::MiniblockNumber;
+use crate::processor::snapshot::types::{MiniblockNumber, StorageValue};
 
-// NOTE: What file extension to use?
-const DEFAULT_EXPORT_PATH: &str = "snapshot_export";
+const DEFAULT_DB_PATH: &str = "snapshot_db";
 
-pub struct SnapshotExporter {
-    storage_log_entries: HashMap<StorageKey, SnapshotStorageLog>,
-    factory_deps: Vec<SnapshotFactoryDependency>,
-    index_to_key_map: IndexSet<U256>,
-    path: PathBuf,
+pub struct SnapshotBuilder {
+    database: SnapshotDB,
 }
```

Comment:
I renamed & reused this as I envisioned that we would build a separate `SnapshotExporter` that would then work from a separate CLI command invocation, open the intermediate RocksDB in read-only mode and just run iterator over it while chunking the output.

This shall wait though, until we receive the protobuf descriptors for the file format.

Reply:
Sure, works for me